appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChart.java

```diff
@@ -80,4 +105,28 @@ public void onPropertyChange(String propertyName, String newValue) {
             setBackgroundColorProperty(newValue);
         }
     }
+
+    /**
+     * Creates a Chart Model instance of the proper type for this Chart.
+     *
+     * @return  New Chart Model instance.
+     */
+    public abstract MockChartModel createChartModel();
```

Comment:
An MVC-like pattern is followed in the implementation here.

View - the Chart itself. Responsible for GUI of the Chart.
Model - an encapsulation of the Data Series object. Handles data logic.
Controller - processes User requests, refreshes Chart, typically calls methods to Model

Is this approach okay to use, or are there other preferences? It seemed to work well in practice so far, provides quite ta nice way to separate some functionality away in 3 classes. It is also used for the Android implementation.

For more information, see the design document section Implementation Design -> Chart & Data Linking